src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java

```diff
@@ -0,0 +1,45 @@
+package edu.oakland.production.middleware01;
+
+import edu.oakland.helper.admin.Satellite;
+import edu.oakland.production.middleware01.MiddlewareGisInterface;
+
+/**
+ * This class shall recieve that GPS signal from display and pass it to GIS processor.
+ *
+ */
+
+public class MiddlewareGisInterfaceImplementation implements MiddlewareGisInterface {
+
+  private MiddlewareGisProcessor middlewareGisProcessor;
+
+  /**
+   * Creates a MiddlewareGisInterfaceImplementation to receive the Satellite signal.
+   *
+   * @param middlewareGisProcessor  The class that this class will pass the Satellite signal to.
+   *
+   */
+  public MiddlewareGisInterfaceImplementation(MiddlewareGisProcessor middlewareGisProcessor) {
+    this.middlewareGisProcessor = middlewareGisProcessor;
+    if (this.middlewareGisProcessor == null) {
+      throw new IllegalArgumentException("processor cannot be null");
+    }
```

Comment:
This is fine if you want to leave it, but generally you want to validate the argument coming in, then only set the attribute if it passes validation. If you want to use that design pattern, it would be like this:
```suggestion
    if (middlewareGisProcessor == null) {
      throw new IllegalArgumentException("processor cannot be null");
    }
    this.middlewareGisProcessor = middlewareGisProcessor;
```